### ISSUE TITLE
"integer" and "numeric" type input fields strip non-digit characters on blur

### ISSUE THREAD

**Issue by wodzuu** [jdorn/json-editor#243]
If we have an input field for JSON type `integer` and populate it with value "1abc2" then it's content get's  modified `on blur` to "12". "lalala" is automatically changed to empty text. I think such field should be highlighted as invalid value instead. The same applies to JSON type `numeric` but here it accepts not only digits but also `e` and `E` so "123hello4" is changed to "123e4".

**Comment by leonbaum** [comment]
I agree.  Same thing when the field is marked `integer`, and the user enters a floating point number.  The fractional value is silently dropped.

**Comment by sqpdln** [comment]
yes, this is very weird behaviour. +1
